src/commonMain/kotlin/fr/acinq/lightning/message/OnionMessages.kt

```diff
@@ -24,13 +27,15 @@ object OnionMessages {
 
     sealed class Destination {
         data class BlindedPath(val route: RouteBlinding.BlindedRoute) : Destination()
-        data class Recipient(val nodeId: PublicKey, val pathId: ByteVector?, val padding: ByteVector? = null, val customTlvs: Set<GenericTlv> = setOf()) : Destination()
+        data class Recipient(val nodeId: PublicKey, val pathId: ByteVector?, val isPhoenix: Boolean, val padding: ByteVector? = null, val customTlvs: Set<GenericTlv> = setOf()) : Destination() {
```

Comment:
Same as the previous one, we could change `val nodeId: PublicKey` to directly take an `EncodedNodeId.WithPublicKey` instead of introducing a boolean flag?

Reply:
Done